pkg/controller/policyrecommendation_registrar.go

```diff
@@ -195,5 +222,21 @@ func (controller *PolicyRecommendationRegistrar) SetupWithManager(mgr ctrl.Manag
 			handler.EnqueueRequestsFromMapFunc(enqueueFunc),
 			builder.WithPredicates(createPredicate),
 		).
+		Watches(
+			&source.Kind{Type: &ottoscaleriov1alpha1.PolicyRecommendation{}},
```

Comment:
Why is the registrar watching PolicyRecommendation changes?

Reply:
It's watching on delete ops for the policyreco and requeues the parent deployment/rollout. Any out of band deletions of policyreco will ensure that there's a new policyreco created.